semana20/Projeto_Spotenu/src/data/AlbumDatabase.ts

```diff
@@ -0,0 +1,43 @@
+import { BaseDatabase } from "./BaseDatabase";
+import { Album } from "../model/Album";
+
+export class AlbumDatabase extends BaseDatabase {
+    public static TABLE_NAME: string = "SpotenuAlbum";
+
+    public static TABLE_RELATION: string = "SpotenuGenreToAlbum";
+
+    private toModel(dbModel?: any): Album | undefined {
+        return dbModel && new Album(dbModel.id, dbModel.band_id, dbModel.name)
+    }
+
+    public async createAlbum(album: Album): Promise<void> {
+        await super.connection()
+            .insert({
+                id: album.getId(),
+                band_id: album.getBandId(),
+                name: album.getName()
+            })
+            .into(AlbumDatabase.TABLE_NAME)
+    }
+
+    public async getAlbumById(id: string): Promise<Album | undefined> {
+        const result = await super.connection()
+            .select("*")
+            .from(AlbumDatabase.TABLE_NAME)
+            .where({ id })
+        return this.toModel(result[0])
+    }
+
+    public async relateGenreAlbum(albumId: string, genreList: string[]): Promise<void>{
+        for(const genre of genreList){
+            await this.connection()
+                .insert({
+                   album_id: albumId,
+                   genre_id: genre
+                })
+                .into(AlbumDatabase.TABLE_RELATION)
+        }
```

Comment:
Vou te dar uma dica aqui para melhorar isso!
**Existe** uma forma de fazer uma query só para adicionar mais de um valor no MySQL. 

- Usando o _query builder_, você pode usar o método `.insertBatch()`
- Usando o _raw_, tem essa indicação:
https://stackoverflow.com/questions/5526917/how-to-do-a-batch-insert-in-mysql